runtime/README.md

```diff
@@ -692,107 +708,504 @@ Command Code: `0x434B_4558` ("CKEX")
+| tag size        | u32      |                 |
+| last length     | u32      |                 |
+| last counter    | u[16]    |                 |
+| last GHASH      | u[16]    |                 |
+| last ciphertext | u[16]    |                 |
+
+### CM_AES_GCM_DECRYPT
+
+Currently only supports AES-256-GCM with a 96-bit IV.
+
+Command Code: `0x434D_4744` ("CMGD")
+
+*Table: `CM_AES_GCM_DECRYPT` input arguments*
+| **Name**        | **Type** | **Description**                          |
+| --------------- | -------- | ---------------------------------------- |
+| chksum          | u32      |                                          |
+| flags           | u32      | Bit 0 = this is the final message        |
+| key CMID        | u8[32]   | CMID of the key to use                   |
+| tag size        | u32      | Number of bytes in the tag.              |
+|                 |          | Can be 0, 1, ..., 16                     |
+|                 |          | Must be 0 if this is not the first block |
+| tag             | u8[...]  |                                          |
+| context size    | u32      | Size of the context                      |
+| context         | u8[...]  |                                          |
+| iv size         | u32      | 12 (first message) or 0 (subsequent)     |
+| iv              | u8[...]  |                                          |
+| aad size        | u32      | Additional authenticated data size       |
+|                 |          | Must be 0 if this is not the first block |
+| aad             | u8[...]  | Additional authenticated data            |
+| ciphertext size | u32      |                                          |
+| ciphertext      | u8[...]  | Data to decrypt                          |
+
+*Table: `CM_AES_GCM_DECRYPT` output arguments*
+| **Name**       | **Type** | **Description**                           |
+| -------------- | -------- | ----------------------------------------- |
+| chksum         | u32      |                                           |
+| fips_status    | u32      | FIPS approved or an error                 |
+| context size   | u32      | Can be 0 if this is the final message     |
+| context        | u8[...]  |                                           |
+| plaintext size | u32      | Can be 0 if this is not the final message |
+| plaintext      | u8[...]  |                                           |
+
+The encrypted and authenticated context's internal structure will be:
+
+*Table: `CM_AES_GCM_DECRYPT` internal context*
+| **Name**        | **Type** | **Description** |
+| --------------- | -------- | --------------- |
+| tag size        | u32      |                 |
+| last length     | u32      |                 |
+| last iv         | u[16]    |                 |
+| last GHASH      | u[16]    |                 |
+| last ciphertext | u[16]    |                 |
+
+
+### CM_ECDH_GENERATE
+
+This computes the first half of an Elliptic Curve Diffie-Hellman exchange.
```

Comment:
Probably worth specifying that this generates an ephemeral ECDH keypair, and the private portion is represented in `context`.